src/components/AnonymousChat/BaseChatItem.tsx

```diff
@@ -1,54 +1,96 @@
+const styles = StyleSheet.create({
+  containerPicture: {
+    height: dimen.normalizeDimen(24),
+    width: dimen.normalizeDimen(24),
+    borderRadius: dimen.normalizeDimen(12)
+  }
+});
+
+const BaseChatItem = ({item, index, type}: BaseChatItemComponentProps) => {
+  const {selectedChannel, handleUserName} = useChatScreenHook(type || ANONYMOUS);
+  const {anon_user_info_emoji_code, anon_user_info_color_code} =
     selectedChannel?.rawJson?.channel || {};
+  const renderAnonymousIcon = (anonColor: string, anonEmoji: string) => (
+    <AnonymousIcon color={anonColor} emojiCode={anonEmoji} size={dimen.normalizeDimen(24)} />
+  );
+
+  const renderDefaultIcon = () => (
+    <FastImage style={styles.containerPicture} source={{uri: item?.user?.profilePicture}} />
+  );
+
+  const handleImageUser = () => {
+    if (type === ANONYMOUS) {
+      if (item?.user?.username !== ANONYMOUS_USER) {
+        return renderDefaultIcon();
+      }
+      return renderAnonymousIcon(anon_user_info_color_code, anon_user_info_emoji_code);
+    }
+
+    if (item?.rawJson?.anon_user_info_emoji_code) {
+      return renderAnonymousIcon(anon_user_info_color_code, anon_user_info_emoji_code);
+    }
+    if (item?.rawJson?.user?.name === ANONYMOUS_USER) {
+      return renderAnonymousIcon(
+        selectedChannel?.rawJson?.anon_user_info_color_code,
+        selectedChannel?.rawJson?.anon_user_info_emoji_code
+      );
+    }
+    return renderDefaultIcon();
+  };
+
+  if (item?.type === 'system' || item?.rawJson?.isSystem || item?.rawJson?.message?.isSystem) {
+    return <BaseSystemChat item={item} index={index} />;
+  }
 
   if (item?.isMe)
     return (
       <ChatItemMyTextV2
-        AnonymousImage={
-          <View style={{marginLeft: 8}}>
-            <AnonymousIcon
-              color={anon_user_info_color_code}
-              emojiCode={anon_user_info_emoji_code}
-              size={24}
-            />
-          </View>
-        }
+        key={index}
+        AnonymousImage={handleImageUser()}
         avatar=""
         isContinuous={item?.isContinuous}
         message={item?.message}
         time={calculateTime(item?.updatedAt, true)}
-        username={`Anonymous ${anon_user_info_emoji_name}`}
+        username={handleUserName(item)}
         type={BaseChatItemTypeProps.MY_ANON_CHAT}
         status={item?.status as ChatStatus}
+        chatType={type}
       />
     );
 
   return (
     <ChatItemTargetText
-      avatar={item?.user?.profilePicture}
+      key={index}
+      AnonymousImage={handleImageUser()}
       isContinuous={item?.isContinuous}
       message={item?.message}
       time={calculateTime(item?.updatedAt, true)}
-      username={item?.user?.username}
+      username={handleUserName(item)}
       type={BaseChatItemTypeProps.ANON_CHAT}
     />
   );
 };
 
-export default BaseChatItem;
+const isEqual = (prevProps, nextProps) => {
+  return nextProps.item === prevProps.item;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The comparison function for `React.memo` has been simplified to only compare the `item` prop. This change could potentially improve performance by reducing unnecessary re-renders, but it might also introduce bugs if other props affect the rendering of the component. Please ensure that this simplification is safe and does not omit necessary checks.

```diff
+ const isEqual = (prevProps, nextProps) => {
+   return nextProps.item === prevProps.item;
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->